.github/workflows/integration-tests.yml

```diff
@@ -26,6 +26,8 @@ jobs:
           key: ${{ runner.os }}-jdk-${{ matrix.java }}-maven-${{ hashFiles('**/pom.xml') }}
           restore-keys: |
             ${{ runner.os }}-jdk-${{ matrix.java }}-maven-
+      - run: mvn -B package -Pdocker-build
+      - run: mvn -B test -Dtest=DebianITCase,DebianGpgITCase,BinaryITCase,RpmITCase
```

Comment:
@olenagerasimova integration tests are started by failsafe plugin in `verify` phase, not `test`, and `verify` test depends on `package` phase, so you don't need to call `package` explicitly: https://maven.apache.org/guides/introduction/introduction-to-the-lifecycle.html
Also, `docker-build` profile is conditionally activated if Docker socket file is available (`/var/run/docker.sock`), so you can omit enabling profile instruction.

Reply:
@g4s8 hm, tests are executed before docker image is build, so it does not work. Also, these commands are temporary, we should rewrite the script when all the tests will be migrated to `TestDeployment`, thus I do not see any point in making it perfectly correct right now

Reply:
@olenagerasimova docker image is built on `package` phase but integration tests are started on `verify` (depends on `package`). The lifecicle is `validate` -> `compile` -> `test` (only unit tests with surefire) -> `package` (creates docker image) -> `verify` (failsafe integration tests). Failsafe plugin is running only on `verify` phase, not on `test`, so it doesn't make sense to specify `*ITCase` classes for `test` phase. All unit tests doesn't depend on docker image, so it's safe to run it before `package` phase. Current config: `mvn -B test -Dtest=DebianITCase,DebianGpgITCase,BinaryITCase,RpmITCase` doesn't start any integration tests, only unit tests